PMax Assets: Tweak copy and add the optional labels

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR implements a part of 📌 [Assets form and related components](https://github.com/woocommerce/google-listings-and-ads/issues/1787#fe-misc) in #1787.

- Align with the finalized copy.
- Add the missed optional labels for the Display URL Path and Portrait images.
- Adjust the layout and style for the help content of the asset fields.
- Adjust the subheading of all marketing images to show the required minimum number only.
- Adjust the help content in the popover of asset images to show different copies according to whether the max number is shared.

### Screenshots:

#### 📷 The optional label

![image](https://user-images.githubusercontent.com/17420811/218401183-a2017fb6-98c8-4de2-9838-e00c116d5318.png)

#### 📷 The help content

![image](https://user-images.githubusercontent.com/17420811/220069257-10b90fe1-22ef-4bcb-aff6-4c208aae32f0.png)
![image](https://user-images.githubusercontent.com/17420811/220069525-6862c67f-b8e2-47ac-ba12-4f31e3c1ea38.png)

### Detailed test instructions:

💡 The Figma link of the finalized copy is the **i2 final** version in the Links section of https://github.com/woocommerce/google-listings-and-ads/issues/1787.

💡 Adjusting to the shared maximum number and the tip floating on the disabled "Add image" button will be done in another PR.

1. Go to step 2 of the campaign creation or editing page.
2. Check if the "(Optional)" label is shown on the fields of Display URL Path and Portrait images.
3. Check if the copy is the same as in the design document.

### Changelog entry

eason9487 committed:
Align with the finalized copy.

eason9487 committed:
Add a new prop `markOptional` to AssetField component.

eason9487 committed:
Mark the optional label for the optional assets in the AssetGroupCard.

eason9487 committed:
Adjust the layout and style for the help content of the asset fields.

eason9487 committed:
Split the `ASSET_IMAGE_SPECS` according to their group.

eason9487 committed:
Add the wording of the shared max number of asset images to `getSubheading` in asset specs.

eason9487 committed:
Add the wording of the shared max number of asset images to `getImageHelpContent` in asset specs.

eason9487 commented on js/src/components/paid-ads/assetSpecs.js:
```diff
@@ -260,56 +271,122 @@ const ASSET_TEXT_SPECS = [
-			imageConfig.suggestedHeight,
-			imageConfig.minWidth,
-			imageConfig.minHeight
+			20,
```
This hard-coded number will be replaced with a variable in the subsequent PR.

puntope commented:
In FAQS-

Second paragraph in the second FAQ is not the same as the provided figma link:

--> GLA

Compared to product ads—which showcase individual products and are designed to drive direct sales and revenue—responsive ads with dynamic ad assets are typically used to highlight your business, generate interest, and attract new customers. While both types of ads can drive conversions, using them together can generate even greater results.

--> FIGMA

Compared to product ads—which showcase individual products and are designed to drive direct sales and revenue— ads with dynamic assets are typically used to highlight your business, generate interest, and attract new customers. While both types of ads can drive conversions, using them together can generate even greater results.


Notice the word "responsive" that is present in GLA but not in FIGMA

puntope commented:
In final URL block text is different than FIGMA

GLA 
Choose a page that people reach after clicking your ad. This might be your homepage, or a more specific page.

FIGMA
Choose a page that you want people to reach after clicking your ad. This might be your homepage, or a more specific page.

puntope commented:
In the text on the side:

GLA
Create ads that effectively boost visibility and generate maximum conversions. Google will mix and match assets to create optimized ads— maximizing your campaign’s performance.

FIGMA
Create ads that effectively boost visibility and generate maximum conversions. Google will mix and match assets to create optimized ads in a variety of formats— maximizing your campaign’s performance.

puntope commented:
In Figma it's says it changed the loading state (I guess now is in the Select button)

<img width="1157" alt="Screenshot 2023-02-21 at 17 24 09" src="https://user-images.githubusercontent.com/5908855/220356646-106d3e4d-07a9-4531-a653-80326252d9ea.png">

puntope commented on js/src/components/paid-ads/assetSpecs.js:
```diff
@@ -260,56 +271,122 @@ const ASSET_TEXT_SPECS = [
 {
-	function getSubheading( min, max ) {
+	function getSubheading( spec, shownAsSharedMax ) {
+		if ( shownAsSharedMax ) {
```
❓ I don't understand well this `shownAsSharedMax` since we are not handling MAX but yes only MIN label, right?  ... Shouldn't be `showOnlyMinLabel` or something like that? 

Another option would be a negation of `showMaxLabel` 

`if ( ! showMaxLabel ) { `

Also, I take as granted that there is always a `spec.max` defined.

puntope replied:
This is 💅  not a blocker

puntope replied:
Ok. I see this name is potentially coming from https://github.com/woocommerce/google-listings-and-ads/pull/1894 shared image max limits. Makes sense now.

eason9487 replied:
Sidenote: The reason is also shown here:
https://github.com/woocommerce/google-listings-and-ads/blob/b9b6b451dd593fba975c3eb5b0b551190bdfece8/js/src/components/paid-ads/assetSpecs.js#L375-L379

So there are three different copies on UI:

![2023-02-22 16 36 52](https://user-images.githubusercontent.com/17420811/220568266-02b2ff2c-ea3e-450a-8c40-a5d4408f5b72.png)

eason9487 committed:
Update copies for assets-related components.

Address
- https://github.com/woocommerce/google-listings-and-ads/pull/1882#issuecomment-1438475477
- https://github.com/woocommerce/google-listings-and-ads/pull/1882#issuecomment-1438480366
- https://github.com/woocommerce/google-listings-and-ads/pull/1882#issuecomment-1438482649

eason9487 commented:
@puntope, thanks for the review and catching the inconsistent copies.

Looks like they have further updates after this PR is opened. The three inconsistencies https://github.com/woocommerce/google-listings-and-ads/pull/1882#issuecomment-1438475477, https://github.com/woocommerce/google-listings-and-ads/pull/1882#issuecomment-1438480366, and https://github.com/woocommerce/google-listings-and-ads/pull/1882#issuecomment-1438482649 were updated in https://github.com/woocommerce/google-listings-and-ads/pull/1882/commits/0fa04f9c46e84782aef48ab30e0ba949363a23bc.

> In Figma it's says it changed the loading state (I guess now is in the Select button)

https://user-images.githubusercontent.com/17420811/220571085-ec6c4587-1ab0-4c19-8a90-5c949e8e0737.mp4

Do you mean that the loading state of the suggestion list behind the "Select final URL" title should be merged into or should replace by the loading state of this button?

After revisiting the conversation with the designer, I think they meant to reword the texts of this button only.

---

Could you help with a new round?

puntope reviewed (approved):
LGTM ✅
